python/cudf_polars/cudf_polars/experimental/benchmarks/tpch.py

```diff
@@ -0,0 +1,1196 @@
+            .group_by("cntrycode")
+            .agg(
+                pl.col("c_acctbal").count().alias("numcust"),
+                pl.col("c_acctbal").sum().round(2).alias("totacctbal"),
+            )
+            .sort("cntrycode")
+        )
+
+
+def _query_type(query: int | str) -> list[int]:
+    if isinstance(query, int):
+        return [query]
+    elif query == "all":
+        return list(range(1, 23))
+    else:
+        return [int(q) for q in query.split(",")]
+
+
+parser = argparse.ArgumentParser(
+    prog="Cudf-Polars TPC-H Benchmarks",
+    description="Experimental Streaming-Executor benchmarks.",
+)
+parser.add_argument(
+    "query",
+    type=_query_type,
+    help="Query number.",
+)
+parser.add_argument(
+    "--path",
+    type=str,
+    default=os.environ.get("TPCH_DATASET_PATH"),
+    help="Root TPCH-dataset directory path.",
+)
+parser.add_argument(
+    "--suffix",
+    type=str,
+    default=".parquet",
+    help="Table file suffix.",
+)
+parser.add_argument(
+    "-e",
+    "--executor",
+    default="dask",
+    type=str,
+    choices=["in-memory", "streaming", "cpu"],
+    help="Executor.",
+)
+parser.add_argument(
+    "-s",
+    "--scheduler",
+    default="synchronous",
+    type=str,
+    choices=["synchronous", "distributed"],
+    help="Scheduler to use with the 'streaming' executor.",
+)
+parser.add_argument(
+    "--n-workers",
+    default=1,
+    type=int,
+    help="Number of Dask-CUDA workers (requires 'distributed' scheduler).",
+)
+parser.add_argument(
+    "--blocksize",
+    default=1000**3,
+    type=int,
+    help="Approx. partition size.",
+)
+parser.add_argument(
+    "--trials",
+    default=1,
+    type=int,
+    help="Number of times to run the query.",
+)
+parser.add_argument(
+    "--debug",
+    default=False,
+    action="store_true",
+    help="Debug run.",
+)
+parser.add_argument(
+    "--shuffle",
+    default=None,
+    type=str,
+    choices=[None, "rapidsmpf", "tasks"],
+    help="Shuffle method to use for distributed execution.",
+)
+parser.add_argument(
+    "--broadcast-join-limit",
+    default=None,
+    type=int,
+    help="Set an explicit `broadcast_join_limit` option.",
+)
+parser.add_argument(
+    "--threads",
+    default=1,
+    type=int,
+    help="Number of threads to use on each GPU.",
+)
+parser.add_argument(
+    "--rmm-pool-size",
+    default=0.5,
+    type=float,
+    help="RMM pool size (fractional).",
+)
+parser.add_argument(
+    "-o",
+    "--output",
+    type=argparse.FileType("at"),
+    default="tpch_results.jsonl",
+    help="Output file path.",
+)
+parser.add_argument(
+    "--summarize",
+    action=argparse.BooleanOptionalAction,
+    help="Summarize the results.",
+    default=True,
+)
+parser.add_argument(
+    "--print-results",
+    action=argparse.BooleanOptionalAction,
+    help="Print the query results",
+    default=True,
+)
+args = parser.parse_args()
+
+
+def run(args: argparse.Namespace) -> None:
+    """Run the benchmark once."""
```

Comment:
```suggestion
    """Run the benchmark."""
```

Reply:
I think "once" is confusing, because `--trials` (which I've suggested renaming to `--iterations`) contradicts that.